tests/metrics_test.py

```diff
@@ -0,0 +1,150 @@
+# This file is part of sbi, a toolkit for simulation-based inference. sbi is licensed
+# under the Affero General Public License v3, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+import numpy as np
+import pytest
+import torch
+from sklearn.neural_network import MLPClassifier
+from torch.distributions import MultivariateNormal as tmvn
+
+from sbi.utils.metrics import c2st, c2st_scores
+
+## c2st related:
+## for a study about c2st see https://github.com/psteinb/c2st/
+
+
+def test_same_distributions(set_seed):
```

Comment:
these six tests all repeat a lot of code! I think it would make sense to move them into two or even one single tests and parametrize that. e.g., 
```python
@pytest.mark.parametrize("dist1, dist2, classifier, target_c2st", 
[
     (tmvn(...), tvm(...), "rf", 0.5), 
     (...),
     ...
],
)
def test_c2st_with_different_distributions(dist1, dist2, classifier, target_c2st, set_seed):
     ndim = 10
     nsamples = 1024

     obs_c2st = c2st(dist1.sample((nsamples,)), dist2.sample((nsamples,)), classifier=classifier, seed=set_seed)

     assert obs_c2st != None
     tol = 0.05
     assert (target_c2st - tol) < obs_c2st[0] < (target_c2st+tol)  # only by chance we differentiate the 2 
```

Reply:
True, but how would I deal with the slow tag then?

Reply:
Yes, I mean `pytest.mark.slow` - which I'd have to use for the MLP tests. 

I am happy to look into that, but in my experience sometimes less generalization is better sometimes as it makes the code more readable. And yes, I am aware that less code is always better. But this change sends me into another loop. :weary:

Reply:
OK, feel free to leave it as it is and add a TODO comment, then we will generalise it later.

Reply:
done with d456162